### PR TITLE
[add] ledger_app.toml C manifest

### PR DESCRIPTION
**Pull request by lpascal-ledger**

### CODE REVIEW

File: ledger_app.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[c-app]
+makefile-path = "./Makefile"
+tests-path = "./tests"
```
Comment (xchapron-ledger) [inline ledger_app.toml:3]: What do you think of moving it in another section ? Like [ragger-tests]?
As rust-app are maybe also going to have ragger tests.

Comment (lpascal-ledger) [reply ledger_app.toml:3]: Another section makes sense indeed. But `ragger-tests` feels too specific, there could be unitest, zemu tests, ...
Also I was under the impression that app mostly had a single `tests` directory, then inside `tests/functional`, `tests/unit`, ... I'm surprised boilerplate has multiple tests directories at the root of the repo. 

I propose (values specific to boilerplate):
```
[tests]
ragger-dir = "./tests/"
unit-dir = "./unit-tests"
```

**Review by xchapron-ledger — COMMENTED** [review]
I'd say part of the objective here is to get rid of some specific settings everywhere.

Another one that is present at multiple location is the list of compatible devices, maybe we can add it there?

For the test part, I'm not sure, I think both @agrojean-ledger and @sgliner-ledger had some usage, and therefore could have needed fields here?
I'm thinking for example at the `flags` input argument of the reusable_build workflow.

**Comment by lpascal-ledger** [comment]
@xchapron-ledger I'm creating an issue on the [workflow repo](https://github.com/LedgerHQ/ledger-app-workflows/issues) as I feel it's the more relevant place for such discussion.

**Comment by agrojean-ledger** [comment]
@lpascal-ledger @xchapron-ledger I guess we either need to change the name of this manifest file or we need to update how we detect rust apps in various scripts (I have `deployer.py` and the reusable workflows in mind) which just look for this `ledger_app.toml`

File: ledger_app.toml (hunk 1 above)
Comment (agrojean-ledger) [reply ledger_app.toml]: I think this [tests] section could be useful to know what kind of functional tests we have : 

* ragger (or maybe just 'pytest' ?)
* zemu (or just 'node' ?)

That would be helpful to tools that automate running those tests. 

Of course we should have a field that gives the location of the tests as well.

**Comment by lpascal-ledger** [comment]
@agrojean-ledger see [here](https://github.com/LedgerHQ/ledger-app-workflows/issues/41) for discussion.

File: ledger_app.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[app]
+build_directory = ""
```
Comment (fbeutin-ledger) [inline ledger_app.toml:2]: "." ?

**Review by xchapron-ledger — APPROVED** [review]

**Pull request merged**